tests/test_video.py

```diff
@@ -126,11 +126,42 @@ def test_fps(self, reset_db: None) -> None:
         assert num_frames_10.count() == 10
         assert num_frames_50.count() == 50
         assert num_frames_1000.count() == 449
-        with pytest.raises(pxt.Error) as exc_info:
+
+        with pytest.raises(pxt.Error, match='At most one of'):
             _ = pxt.create_view(
                 'invalid_args', videos, iterator=FrameIterator.create(video=videos.video, fps=1 / 2, num_frames=10)
             )
-        assert 'At most one of `fps` or `num_frames` may be specified' in str(exc_info.value)
+
+    def test_keyframes_only(self, reset_db: None) -> None:
+        path = get_video_files()[0]
+        videos = pxt.create_table('videos', {'video': pxt.Video})
+
+        # Test keyframes_only=True extracts all keyframes
+        keyframes = pxt.create_view(
+            'keyframes',
+            videos,
+            iterator=FrameIterator.create(video=videos.video, keyframes_only=True, all_frame_attrs=True),
+        )
+        frames = pxt.create_view(
+            'frames', videos, iterator=FrameIterator.create(video=videos.video, fps=0, all_frame_attrs=True)
+        )
+
+        videos.insert(video=path)
+
+        # Verify keyframes were extracted
+        keyframes_count = keyframes.count()
+        res = frames.order_by(frames.pos).collect()
+        assert keyframes_count == sum(int(attrs['key_frame']) for attrs in res['frame_attrs'])
```

Comment:
The `int()` is extraneous here (`sum(attrs['key_frame']) for attrs in ...` is sufficient)

Reply:
Done.